back/app/models/notifications/project_phase_upcoming.rb

```diff
@@ -76,10 +76,9 @@ def self.make_notifications_on(activity)
       end
     end
 
-    private
-
-    def recipients(_project_id)
+    def self.recipients(phase)
       UserRoleService.new.moderators_for(phase)
     end
+    private_class_method :recipients
```

Comment:
Instead of using a private class method which holds one line, I would suggest to inline the method in `make_notifications_on`, like this:
```
if phase.project
  recipients = UserRoleService.new.moderators_for(phase)
  recipients.ids.map do |recipient_id|
```
That results in less code, and less complexity for specifying that the method is private.

Reply:
Yes, makes sense :)